### DISCUSSION TITLE
Understand transformers in saved pipeline

### DISCUSSION THREAD

**Discussion by DipanjanSanyal** [pycaret/pycaret#2641] [Q&A] [answered]
I have experimented and saved a regression model pipeline using pycaret. Then I am loading it, making predictions and trying to explain the steps to devops team.

```
< many steps doing data download and custom data preparation >
model = load_model('saved pipeline')
data_with_pred = predict_model(model)
< few steps doing post-hoc calculations and data upload >

model3.named_steps
```

The transformers in the pipeline, except for the model, are not found in scikit-learn documentation. For example DataTypes_Auto_infer(). Where can I see the source code for these transformers?

**Comment by ngupta23** [comment, marked as answer]
Yes, in pycaret 2.3.x, some of the transformers are not scikit compatible. This should have been rectified with the recently released version of pycaret 3.0.0rc. You can try it here:

`pip install -U --pre pycaret`

**Reply by DipanjanSanyal** [reply]
Really appreciate!
Any way to dig into those transformer codes for 2.3.x? My situation is that I already have the pkl saved using 2.3.10 and I cannot reproduce the training data.
